src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -1556,16 +1556,20 @@ public bool IsExecutingInvocation(string invocationId)
 
         public bool TryFailExecutions(Exception workerException)
         {
-            if (workerException == null)
-            {
-                return false;
-            }
-
             foreach (var invocation in _executingInvocations?.Values)
             {
                 string invocationId = invocation.Context?.ExecutionContext?.InvocationId.ToString();
                 _workerChannelLogger.LogDebug("Worker '{workerId}' encountered a fatal error. Failing invocation: '{invocationId}'", _workerId, invocationId);
-                invocation.Context?.ResultSource?.TrySetException(workerException);
+
+                if (workerException is not null)
+                {
+                    invocation.Context?.ResultSource?.TrySetException(workerException);
+                }
+                else
+                {
+                    invocation.Context?.ResultSource?.TrySetCanceled();
+                }
+
                 RemoveExecutingInvocation(invocationId);
             }
             return true;
```

Comment:
When will this function ever return false? Is there is no scenario for it, why is it a bool?

Reply:
Looked into this - the bool returned also isn't used anywhere (in all references, this method is called with no action taken based on the result). Will refactor.